### PR TITLE
Initial layout

### PR DESCRIPTION
**Pull request by burcinkarimov**

### CODE REVIEW

File: src/components/PrimaryButton.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+function PrimaryButton({ text }) {
+  return(
+    <button type="submit" className="px-4 py-2 w-20 border-[1px] bg-indigo-100 hover:bg-indigo-300">{text}</button>
```
Comment (ideriabin) [inline src/components/PrimaryButton.js:3]: small suggestion: `type="submit"` can be omitted here

the current default behavior for `<button>` is that it behaves like `type="submit"` inside the form and does nothing everywhere else

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#type

**Comment by ideriabin** [comment]
Great work! Now let's try making `App.js` less heavy — currently you have all the layout defined in there, and as your application grows it will grow only larger. Try thinking about what parts you can split the UI into, and define components representing these parts of the UI.

Let's define the list of the parts/components for the UI here and then implement them as a part of this PR — this will be the final state of our initial layout.

**Review by ideriabin — CHANGES_REQUESTED** [review]

**Comment by burcinkarimov** [comment]
> Great work! Now let's try making `App.js` less heavy — currently you have all the layout defined in there, and as your application grows it will grow only larger. Try thinking about what parts you can split the UI into, and define components representing these parts of the UI.
> 
> Let's define the list of the parts/components for the UI here and then implement them as a part of this PR — this will be the final state of our initial layout.

So the components that I’ve thought to divide the app into are:

- Categories -> dedicated for the block that will hold category list and also the form to add new category
    - AddNewCategory
    - CategoriesList  -> for displaying the list of categories
        - CategoriesListItem -> for one item in the list of categories
        
- Dashboard -> for the whole block that will hold tasks list, search bar, show done, and form for adding new task
    - TaskControlPanel -> for the top section of dashboard, displays search bar, show done, add new task
        - SearchBar
        - ShowDone
        - AddNewTask
    - TasksList -> list of tasks
        - TasksListItem -> one task item 

AddNewCategory and AddNewTask are very similar so they could be the same component, but also seems better to me to have them separate since there are just two of them and not more, because I remember seeing that you apply DRY starting from three similar components.

**Comment by ideriabin** [comment]
> So the components that I’ve thought to divide the app into are:
> 
> * Categories -> dedicated for the block that will hold category list and also the form to add new category
>   
>   * AddNewCategory
>   * CategoriesList  -> for displaying the list of categories
>     
>     * CategoriesListItem -> for one item in the list of categories
> * Dashboard -> for the whole block that will hold tasks list, search bar, show done, and form for adding new task
>   
>   * TaskControlPanel -> for the top section of dashboard, displays search bar, show done, add new task
>     
>     * SearchBar
>     * ShowDone
>     * AddNewTask
>   * TasksList -> list of tasks
>     
>     * TasksListItem -> one task item
> 
> AddNewCategory and AddNewTask are very similar so they could be the same component, but also seems better to me to have them separate since there are just two of them and not more, because I remember seeing that you apply DRY starting from three similar components.

Great job! `AddNewCategory` and `AddNewTask` should be separate indeed — the may look identical right now, but their context is different, and their design and functionality may change in the future.

Now let's implement this component structure in the scope of this PR

**Review by ideriabin — APPROVED** [review]

**Pull request merged**